azure-samples/src/main/java/com/microsoft/azure/management/compute/samples/ManageResourceFromMSIEnabledVirtualMachineBelongsToAADGroup.java

```diff
@@ -0,0 +1,210 @@
+import com.microsoft.azure.management.storage.StorageAccount;
+import com.microsoft.rest.LogLevel;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Azure Compute sample for managing virtual machines -
+ *   - Create a AAD security group
+ *   - Assign AAD security group Contributor role at a resource group
+ *   - Create a virtual machine with MSI enabled
+ *   - Add virtual machine MSI service principal to the AAD group
+ *   - Set custom script in the virtual machine that
+ *          - install az cli in the virtual machine
+ *          - uses az cli MSI credentials to create a storage account
+ *   - Get storage account created through MSI credentials.
+ */
+public final class ManageResourceFromMSIEnabledVirtualMachineBelongsToAADGroup {
+    /**
+     * Main function which runs the actual sample.
+     *
+     * @param azure instance of the azure client
+     * @return true if sample runs successfully
+     */
+    public static boolean runSample(Azure azure) {
+        final String linuxVMName = Utils.createRandomName("VM1");
+        final String rgName = Utils.createRandomName("rgCOMV");
+        final String pipName = Utils.createRandomName("pip1");
+        String group1Name = SdkContext.randomResourceName("group", 16);
+        String roleAssignmentName = SdkContext.randomUuid();
+        final String userName = "tirekicker";
+        final String password = "12NewPA$$w0rd!";
+        final Region region = Region.US_WEST_CENTRAL;
+
+        final String installScript = "https://raw.githubusercontent.com/Azure/azure-sdk-for-java/master/azure-samples/src/main/resources/create_resources_with_msi.sh";
+        String installCommand = "bash create_resources_with_msi.sh {subscriptionID} {port} {stgName} {rgName} {location}";
+        List<String> fileUris = new ArrayList<>();
+        fileUris.add(installScript);
+
+        try {
+
+            //=============================================================
+            // Create a AAD security group
+
+            System.out.println("Creating a AAD security group");
+
+            ActiveDirectoryGroup activeDirectoryGroup = azure.accessManagement()
+                    .activeDirectoryGroups()
+                    .define(group1Name)
+                    .withEmailAlias(group1Name)
+                    .create();
+
+            //=============================================================
+            // Assign AAD security group Contributor role at a resource group
+
+            ResourceGroup resourceGroup = azure.resourceGroups()
+                    .define(rgName)
+                    .withRegion(region)
+                    .create();
+
+            SdkContext.sleep(45 * 1000);
+
+            System.out.println("Assigning AAD security group Contributor role to the resource group");
+
+            azure.accessManagement()
+                    .roleAssignments()
+                    .define(roleAssignmentName)
```

Comment:
could we make sure we use Fluent-structure respecting indentation?  the flow under define should be one more level of indentation deeper, until (and including) `.create()'

For child definitions, the stuff under `.defineChild()` would similarly have an extra level of indentation, all the way until (and including) `.attach()`